packages/fxa-auth-server/test/local/payments/stripe-product-and-plans-converter.js

```diff
@@ -0,0 +1,536 @@
+      );
+    });
+    it('processes only the product with productId when passed', async () => {
+      await converter.convert({ ...args, productId: product1.id });
+      sinon.assert.calledOnceWithExactly(
+        converter.stripeHelper.stripe.products.list,
+        { ids: [product1.id] }
+      );
+    });
+    it('does not update Firestore if dryRun = true', async () => {
+      const argsDryRun = { ...args, isDryRun: true };
+      await converter.convert(argsDryRun);
+      sinon.assert.notCalled(mockPaymentConfigManager.storeProductConfig);
+      sinon.assert.notCalled(mockPaymentConfigManager.storePlanConfig);
+      sinon.assert.calledWith(
+        mockLog.debug.firstCall,
+        'StripeProductsAndPlansConverter.convertBegin',
+        argsDryRun
+      );
+      sinon.assert.calledWith(
+        mockLog.debug.secondCall,
+        'StripeProductsAndPlansConverter.dryRun.convertPlanSuccess',
+        {
+          stripePlanId: plan1.id,
+          planConfig: planConfig1,
+        }
+      );
+      sinon.assert.calledWith(
+        mockLog.debug.thirdCall,
+        'StripeProductsAndPlansConverter.dryRun.convertPlanSuccess',
+        {
+          stripePlanId: plan2.id,
+          planConfig: planConfig2,
+        }
+      );
+      sinon.assert.calledWith(
+        mockLog.debug.getCall(3),
+        'StripeProductsAndPlansConverter.dryRun.convertProductSuccess',
+        {
+          stripeProductId: product1.id,
+          productConfig: productConfig1,
+        }
+      );
+    });
+    it('moves localized data from plans into the productConfig', async () => {
+      // TODO: Use a real PaymentsConfigManager everywhere? Would probably simplify things.
+      // For this test we create a real PaymentConfigManager (with a Firestore emulator),
+      // so we can confirm that its cached `this.products` has a productConfig
+      // with the value of the `locales` field derived from the plans.
+      const mockConfig = {
+        authFirestore: {
+          prefix: 'mock-fxa-',
+        },
+        subscriptions: {
+          playApiServiceAccount: {
+            credentials: {
+              clientEmail: 'mock-client-email',
+            },
+            keyFile: 'mock-private-keyfile',
+          },
+        },
+      };
+      const firestore = setupFirestore(mockConfig);
+      Container.set(AuthFirestore, firestore);
+      Container.set(AuthLogger, mockLog);
+      Container.set(AppConfig, mockConfig);
+      const paymentConfigManager = new PaymentConfigManager();
+      Container.set(PaymentConfigManager, paymentConfigManager);
+      const productConfigDbRef = paymentConfigManager.productConfigDbRef;
+      const planConfigDbRef = paymentConfigManager.planConfigDbRef;
+      await deleteCollection(
+        paymentConfigManager.firestore,
+        productConfigDbRef,
+        100
+      );
+      await deleteCollection(
+        paymentConfigManager.firestore,
+        planConfigDbRef,
+        100
+      );
```

Comment:
If I remove this; the `await retry` further down times out. I haven't figured out why yet.

Reply:
It's because I wasn't restarting the Firestore emulator between tests. Thanks Barry!